PIPELINE.md

```diff
@@ -0,0 +1,126 @@
+cluster.
+
+## Basic: Getting Started
+
+1. Start an OpenShift cluster (origin v1.3 and later). Create a new project.
+2. [Optional] Create a fork of this repository
+3. Instantiate the pipeline from template by invoking:
+   ```
+   oc new-app -f https://raw.githubusercontent.com/FORK/parksmap-web/add-jenkinsfile/ose3/pipeline-buildconfig-template.json -p GIT_URI=https://github.com/FORK/parksmap-web.git
+   ```
+
+   (Where FORK is your own fork of this repository or `openshift-roadshow`)
+4. Start the pipeline build from the CLI with:
+   ```
+   oc start-build parksmap-web-pipeline
+   ```
+
+   OR
+
+   Start the pipeline from the web console. Navigate to the project, Builds->Pipelines.
+   Click `Start` next to the pipeline.
+
+
+## Advanced: Example of a PR-based development flow
+
+The Jenkinsfile in this repository can be used to support a PR flow with Jenkins on OpenShift.
+The main application is built and deployed to an 'integration' project. Every time a PR is
+submitted to the source repository, a new OpenShift project is created to build and test the
+PR and allow for manual testing. Once the manual testing is approved, the build succeeds.
+
+The following steps guide you through the setup of the Github Organization Folder plugin in
+Jenkins and the required steps on the GitHub side. The Github Organization Folder plugin
+instantiates a pipeline for every project it finds with a Jenkinsfile at its root. It also
+sets up webhooks in GitHub so that new PRs and changes to a branch are automatically built
+in Jenkins. In order for webhooks to work, the ngrok application is used to create a public
+endpoint for the internal Jenkins service.
+
+### GitHub Setup
+
+1. Create your own forks of [nationalparks](https://github.com/openshift-roadshow/nationalparks.git),
+   [mlbparks](https://github.com/openshift-roadshow/mlbparks.git), and
+   [parksmap-web](https://github.com/openshift-roadshow/parksmap-web.git).
+2. Rename each of those repositories in GitHub with a common prefix, for example:
+   `myapp-nationalparks`, `myapp-mlbparks`, `myapp-parksmap`.
+3. Create a GitHub personal access token with access to `repo`, `admin:repo_hook`, `admin:org_hook`.
+   (Keep your token to use below as GITHUB-TOKEN).
+
+### OpenShift/Jenkins Setup
+
+1. Create a new project:
+   ```
+   oc new-project PROJECT-NAME
+   ```
+
+2. Assign the right permission to your jenkins service account:
+
+   ```
+   oc login -u system:admin
+   oc adm policy add-cluster-role-to-user self-provisioner system:service-account:PROJECT-NAME:jenkins
+   oc login -u developer
+   ```
+
+3. Add Jenkins to your project: 
+
+   ```
+   oc new-app openshift/jenkins-ephemeral -p MEMORY_LIMIT=1Gi
+   ```
+
+4. Expose the jenkins service externally (for GitHub webhooks) with ngrok: 
+
+   ```
+   oc new-app -f https://raw.githubusercontent.com/csrwng/ngrok/master/openshift/ngrok-template.yaml -p HOST=jenkins -p PORT=80
```

Comment:
@csrwng cause in public clusters using ngrok is kind of overkilling and I would rather not use it as preferred way of showing how to use webhooks but rather as alternative in case the cluster is not accessible.